### ISSUE TITLE
Unbiased estimation in batch normalization function

### ISSUE THREAD

**Issue by takerum** [pfnet/chainer#332]
Hi!

I'm now implementing neural network models with chainer.
I have a question regarding in the implementation of batch normalization function in chainer.
Following code is the computation of the average of means and the average of varinaces for the test mode in chainer/functions/batch_normalization.py

            m = ldim * rdim
            adjust = m / max(m - 1., 1.)  # unbiased estimation
            self.avg_mean *= decay
            self.avg_mean += (1 - decay) * adjust * mean
            self.avg_var *= decay
            self.avg_var += (1 - decay) * adjust * var
Why is the estimation of the "avg_mean" scaled by m/(m-1)?
I think only the "avg_var" needs to be scaled by m/(m-1) for unbiased estimation.
In the original paper of batch normalization:http://arxiv.org/pdf/1502.03167v3.pdf, only the average of variances is also scaled by m/(m-1), and the average of means is not scaled. 

Best,

**Comment by beam2d** [comment]
Thank you for reporting the suspicious code! I agree you that the scaling is not needed for avg_mean. It seems a bug, so I will fix it.

**Comment by takerum** [comment]
Thank you very much!

**Comment by takerum** [comment]
I found the codes that might be wrong in batch_normalization.py
```
       if self.is_finetune:
                self.N[0] += 1
                decay = 1. / self.N[0]
            else:
                decay = self.decay

            m = ldim * rdim
            adjust = m / max(m - 1., 1.)  # unbiased estimation
            self.avg_mean *= decay
            self.avg_mean += (1 - decay) * mean
            self.avg_var *= decay
            self.avg_var += (1 - decay) * adjust * var
```
In finetune mode, I think ```decay = 1. / self.N[0]``` is wrong and ```decay = 1 - 1. / self.N[0]``` is correct for finetuning the batch statistics.

Best,
